packages/shopify-codemod/transforms/split-if-statement-assignment-parameter.js

```diff
@@ -0,0 +1,30 @@
+export default function splitIfStatementAssignmentParameter({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.IfStatement, {
+      type: j.IfStatement.name,
+      test: {
+        type: j.AssignmentExpression.name,
+      },
+    }).forEach((nodePath) => {
+      const args = nodePath.get('test').node;
+      nodePath.insertBefore(j.expressionStatement(args));
+      nodePath.get('test').replace(args.left);
+
+    });
+  sourceAST
+    .find(j.IfStatement, {
+      test: {
+        type: j.SequenceExpression.name,
+        expressions: (expressions) => expressions.length > 1,
```

Comment:
This will fail for expressions that are not assignment expressions (`if (foo, bar) {}`). Maybe add a test for `If (foo = bar, baz) {}`, and one for `if (foo = bar, baz = qux) {}` and write the code to make those pass.